src/lib.rs

```diff
@@ -171,7 +171,15 @@ impl<T> core::panic::RefUnwindSafe for Event<T> {}
 
 impl<T> fmt::Debug for Event<T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_str("Event { .. }")
+        let inner = self.inner();
```

Comment:
Instead of calling `inner()`, call `try_inner()`, and if it returns `None` output `Event(<uninitialized>)` using `debug_tuple`. See [here](https://github.com/smol-rs/async-executor/blob/2cfb6e4ed03c7eaf6c5fb2d3170ac870ba0a9b9d/src/lib.rs#L981-L995) for an example.